src/SFA.DAS.EmployerIncentives.Commands/ApprenticeshipIncentive/CalculateDaysInLearning/CalculateDaysInLearningCommandHandler.cs

```diff
@@ -0,0 +1,44 @@
+﻿using SFA.DAS.EmployerIncentives.Abstractions.Commands;
+using SFA.DAS.EmployerIncentives.Commands.Persistence;
+using SFA.DAS.EmployerIncentives.Commands.Services;
+using SFA.DAS.EmployerIncentives.Domain.ApprenticeshipIncentives;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.EmployerIncentives.Commands.ApprenticeshipIncentive.CalculateDaysInLearning
+{
+    public class CalculateDaysInLearningCommandHandler : ICommandHandler<CalculateDaysInLearningCommand>
+    {
+        private readonly IApprenticeshipIncentiveDomainRepository _incentiveDomainRepository;
+        private readonly ILearnerDomainRepository _learnerDomainRepository;
+        private readonly ICollectionCalendarService _collectionCalendarService;
+
+        public CalculateDaysInLearningCommandHandler(
+            IApprenticeshipIncentiveDomainRepository incentiveDomainRepository,
+            ILearnerDomainRepository learnerDomainRepository,
+            ICollectionCalendarService collectionCalendarService)
+        {
+            _incentiveDomainRepository = incentiveDomainRepository;
+            _learnerDomainRepository = learnerDomainRepository;
+            _collectionCalendarService = collectionCalendarService;
+        }
+
+        public async Task Handle(CalculateDaysInLearningCommand command, CancellationToken cancellationToken = default)
+        {
+            var incentive = await _incentiveDomainRepository.Find(command.ApprenticeshipIncentiveId);
+
+            if(incentive == null)
+            {
+                return;
+            }
+
+            Learner learner = await _learnerDomainRepository.GetOrCreate(incentive);
```

Comment:
Should this use GetOrCreate?  If we don't have a learner record we shouldn't get to this point.

Reply:
The ILearnerDomainRepository.GetOrCreate is used elsewhere so it's just being repurposed.

Reply:
Will add another method to the interface

Reply:
Done.